extension/src/shared/layerContainers/buildingModel.tsx

```diff
@@ -134,32 +133,32 @@ export const BuildingModelLayerContainer: FC<TilesetContainerProps> = ({
   const wireframeAtom = useFindComponent(componentAtoms, TILESET_WIREFRAME);
 
   const hiddenFeatures = useAtomValue(hiddenFeaturesAtom);
-  const hiddenFeaturesConditions: ConditionsExpression = useMemo(
-    () => ({
-      conditions:
-        hiddenFeatures?.map(f => [`${variable("gml_id")} ===  ${string(f)}`, "false"]) ?? [],
-    }),
-    [hiddenFeatures],
-  );
+  const hiddenFeaturesConditions: ConditionsExpression = {
+    conditions: [["${hiddenFeatureIds} == ${gml_id}", "false"]],
+  };
 
   const searchedFeatures = useAtomValue(searchedFeaturesAtom);
+
+  const defines: Record<string, string> = {
+    hiddenFeaturesIds: JSON.stringify(hiddenFeatures),
+  };
+
+  if (searchedFeatures?.onlyShow) {
+    defines.searchedFeaturesIds = JSON.stringify(searchedFeatures.features);
+  }
   const shownSearchedFeaturesConditions: ConditionsExpression | undefined = useMemo(
     () =>
       searchedFeatures?.onlyShow
         ? {
             conditions: [
-              ...(searchedFeatures?.conditions?.flatMap(([key, values]) =>
-                values.map(
-                  v => [`${variable(key)} === ${numberOrString(v)}`, "true"] as [string, string],
-                ),
-              ) ?? []),
+              ["${searchedFeaturesIds} == ${gml_id}", "true"],
```

Comment:
`searchedFeatures.conditions` isn't using id, so I think we don't need to use the variable in this case 🙏

Reply:
Ahh, Thanks for pointing that out🙇